library/src/main/java/com/securepreferences/SecurePreferences.java

```diff
@@ -231,6 +236,11 @@ private String generateAesKeyName(Context context, int iterationCount) throws Ge
      */
     @SuppressLint("HardwareIds")
     private static String getDeviceSerialNumber(Context context) {
+        // Android API 28 needs another way to get the serial number. Assume that
+        // the READ_PHONE_STATE permission in AndroidManifest.xml is respected.
+        if (Build.VERSION.SDK_INT >= 28) {
+            return Build.getSerial();
+        }

```

Comment:
This is not safe and will crash the app if the user does not grant the READ_PHONE_STATE permission. You either need to check for permission before calling this, or let the user handle the salt via a constructor like in the PR I submitted here: https://github.com/scottyab/secure-preferences/pull/91

Reply:
Good one, I will add the check. I am also worried about the "unknown" value, is it uppercase or lowercase? I assume we should `Build.UNKNOWN` (https://developer.android.com/reference/android/os/Build.html#UNKNOWN) constant. I could not test it properly. @willswinson

Reply:
It is lowercase and does seem to come from `Build.UNKNOWN`